### ISSUE TITLE
lorax-composer API modules/projects  fail with: Cannot find a valid baseurl for repo: base/$releasever/x86_64

### ISSUE THREAD

**Issue by martinpitt** [rhinstaller/lorax#282]
This is with the latest version 19.7.4-1.el7.centos from the COPR. Reproducer on current CentOS 7 image:
```
yum install -y epel-release
curl https://copr.fedorainfracloud.org/coprs/g/weldr/lorax-composer/repo/epel-7/group_weldr-lorax-composer-epel-7.repo > /etc/yum.repos.d/lorax-composer.repo
yum install -y lorax-composer 
systemctl enable --now lorax-composer
curl http://localhost:4000/api/v0/projects/list
```

This works:
```
# curl http://localhost:4000/api/v0/recipes/list
{
  "limit": 20, 
  "offset": 0, 
  "recipes": [], 
  "total": 0
}
```

This doesn't:
```
[root@m1 ~]# curl http://localhost:4000/api/v0/projects/list
{
  "error": {
    "msg": "('There was a problem listing projects: %s', 'Cannot find a valid baseurl for repo: base/$releasever/x86_64')"
  }

# curl http://localhost:4000/api/v0/modules/list
{
  "error": {
    "msg": "There was a problem listing modules: Cannot find a valid baseurl for repo: base/$releasever/x86_64"
  }
```

Note that I'm not a project member and thus I cannot set labels. Can you please add "lorax-composer"?

**Comment by martinpitt** [comment]
Also, it seems for /projects/list you wrote a `,` instead of a `%` in the error message construction :-)

**Comment by martinpitt** [comment]
/etc/yum.repos.d/CentOS-Base.repo has entries like
```
[base]
name=CentOS-$releasever - Base
mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=os&infra=$infra
#baseurl=http://mirror.centos.org/centos/$releasever/os/$basearch/
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
```
It seems `$basearch` got substituted correctly, but not `$basever`?

**Comment by martinpitt** [comment]
If I do

    sed -i 's/\$releasever/7/g' /etc//yum.repos.d/*.repo
    systemctl restart lorax-composer

then the two API paths start to work. At least I get some names:

    # curl -s http://localhost:4000/api/v0/projects/list|grep '"name"' | wc -l
    20
    # curl -s http://localhost:4000/api/v0/modules/list|grep '"name"' | wc -l
   20

I suppose you deliberately limit the results to 20 for initial testing? Or is this another bug?

**Comment by martinpitt** [comment]
This only happens after doing an /api/v0/modules/list query. Not doing any query or just listing recipes does *not* cause the hang.

**Comment by martinpitt** [comment]
shutdown journal has no useful information from lorax itself, other than SIGTERM just timed out.

This actually reproduces pretty well without a shutdown:
```
# systemctl start lorax-composer
# systemctl stop lorax-composer  # fast
# systemctl start lorax-composer
# curl http://localhost:4000/api/v0/projects/list
# systemctl stop lorax-composer  # hangs; Control-C
# # systemctl status lorax-composer
● lorax-composer.service - Lorax Image Composer API Server
   Loaded: loaded (/usr/lib/systemd/system/lorax-composer.service; enabled; vendor preset: disabled)
   Active: deactivating (stop-sigterm) since Thu 2017-11-23 04:37:25 EST; 37s ago
```

So after processing a request it somehow stops responding to SIGTERM.

**Comment by martinpitt** [comment]
Erk, the two previous comments were supposed to go to  issue #283. Deleting them here and copying them there.

**Comment by bcl** [comment]
> I suppose you deliberately limit the results to 20 for initial testing? Or is this another bug?
Routes that may list large amounts of unexpected data are paged, with the default being 20 items. You can change this by setting limit and offset in the url query.

**Comment by bcl** [comment]
#285 should fix this. You can either pass --releasever or let it use yum to figure it out.

**Comment by martinpitt** [comment]
Confirmed that this is fixed with 0.19.5. Thanks!
